cyto_dl/nn/vits/blocks/patchify/patchify_hiera.py

```diff
@@ -0,0 +1,124 @@
+from typing import List, Optional
+
+import numpy as np
+import torch
+from einops import rearrange, repeat
+from einops.layers.torch import Rearrange
+from timm.models.layers import trunc_normal_
+
+from cyto_dl.nn.vits.blocks.patchify.patchify_base import PatchifyBase
+
+
+def take_indexes_mask(sequences, indexes):
+    """
+    sequences: batch x mask units x patches x emb_dim
+    indexes: mask_units x batch
+    """
+    # always gather across tokens dimension
+    return torch.gather(
+        sequences,
+        1,
+        repeat(
+            indexes,
+            "mu b -> b mu p c",
+            b=sequences.shape[0],
+            c=sequences.shape[-1],
+            mu=sequences.shape[1],
+            p=sequences.shape[2],
+        ),
+    )
+
+
+class PatchifyHiera(PatchifyBase):
+    """Class for converting images to a masked sequence of patches with positional embeddings."""
```

Comment:
to "mask units" instead of masked sequence? since that's what a regular patchify does?